### PR TITLE
Stage 2b PR Train Step 1: shared tab shell + tracker manager migration

### PR DESCRIPTION
**Pull request by davidhouweling**
## 1. context
Stage 2a (backend film integration for kill matrix analytics) was merged in PR #538. Stage 2b moves the frontend toward shared tabs and kill matrix UI across Match Stats, Series Stats, and Individual Tracker paths. This PR is the first step in a PR train to keep each change set small and reviewable.

## 2. overview of the feature
Introduce a reusable tab shell component for pages UI and migrate the Individual Tracker Manager shell to use it. This establishes the shared tab behavior foundation required before wiring kill matrix tabs into all stats surfaces.

## 3. what this PR does
- Adds a reusable `TabbedSection` component:
  - `pages/src/components/shared/tabbed-section/tabbed-section.tsx`
  - `pages/src/components/shared/tabbed-section/tabbed-section.module.css`
  - `pages/src/components/shared/tabbed-section/types.ts`
- Refactors Individual Tracker Manager shell to use `TabbedSection`:
  - `pages/src/components/individual-tracker-manager/individual-tracker.tsx`
- Removes now-unused local tab styles from:
  - `pages/src/components/individual-tracker-manager/individual-tracker.module.css`
- Adds tests for the new shared component and migrated shell:
  - `pages/src/components/shared/tabbed-section/tests/tabbed-section.test.tsx`
  - `pages/src/components/individual-tracker-manager/tests/individual-tracker-shell.test.tsx`

Validation:
- `npx vitest run pages/src/components/shared/tabbed-section/tests/tabbed-section.test.tsx pages/src/components/individual-tracker-manager/tests/individual-tracker-shell.test.tsx pages/src/components/stats/tests/match-stats.test.tsx pages/src/components/stats/tests/series-stats.test.tsx`
- `npm run typecheck:pages`

## 4. PR train
| Step | Branch | PR | Scope | Status |
|---|---|---|---|---|
| 1 | `feature/stage-2b-kill-matrix-ui` | this PR | Shared tab shell + Individual Tracker Manager migration + tests | Open |
| 2 | `feature/stage-2b-analytics-service` | pending | Pages analytics client/service integration + kill matrix presenter/store + tests | Planned |
| 3 | `feature/stage-2b-kill-matrix-table` | pending | Kill matrix table UI and wiring into Match Stats, Series Stats, Individual Tracker Viewer + tests | Planned |

### CODE REVIEW

File: pages/src/components/shared/tabbed-section/tabbed-section.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          const panelDomId = `${tabSetId}-${tab.id}-panel`;
+          const isSelected = tab.id === selectedTabId;
+          return (
+            <button
+              key={tab.id}
+              id={tabDomId}
+              type="button"
+              role="tab"
+              aria-selected={isSelected}
+              aria-controls={panelDomId}
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/shared/tabbed-section/tabbed-section.tsx:98]: Fixed in f74c444. Added roving tabindex so only the selected tab is tabbable, with test coverage for the active/inactive tabIndex states.

File: pages/src/components/shared/tabbed-section/tabbed-section.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+              className={classNames(styles.tabButton, {
+                [styles.tabButtonActive]: isSelected,
+              })}
+              onClick={(): void => {
+                onTabChange(tab.id);
+              }}
+            >
+              {tab.label}
+            </button>
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/shared/tabbed-section/tabbed-section.tsx:107]: Fixed in 7412030. Added ArrowLeft/ArrowRight/Home/End keyboard navigation to the tablist with 5 new tests covering all directions and wraparound.

File: pages/src/components/shared/tabbed-section/tabbed-section.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  selectedTabId,
+  onTabChange,
+  tabListAriaLabel,
+}: TabbedSectionProps<TId>): React.ReactElement {
+  const tabSetId = React.useId();
+
+  const handleKeyDown = React.useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>): void => {
+      const currentIndex = tabs.findIndex((t) => t.id === selectedTabId);
+      let nextIndex: number | null = null;
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/shared/tabbed-section/tabbed-section.tsx:45]: Fixed in 127a885. Added early returns when tabs is empty or selectedTabId is not found, with two new tests covering both edge cases.

File: pages/src/components/shared/tabbed-section/tabbed-section.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        default: {
+          return;
+        }
+      }
+
+      event.preventDefault();
+      onTabChange(Preconditions.checkExists(tabs[nextIndex], "tab at index").id);
+    },
+    [tabs, selectedTabId, onTabChange],
+  );
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/shared/tabbed-section/tabbed-section.tsx:75]: Fixed in 127a885. After key navigation, focusTab() imperatively focuses the newly active tab button using a data-tabid attribute lookup on the tablist ref, with a dedicated test asserting document.activeElement.

File: pages/src/components/shared/tabbed-section/tabbed-section.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  onTabChange,
+  tabListAriaLabel,
+}: TabbedSectionProps<TId>): React.ReactElement {
+  const tabSetId = React.useId();
+  const tabListRef = React.useRef<HTMLDivElement>(null);
+
+  const focusTab = React.useCallback((tabId: TId): void => {
+    const button = tabListRef.current?.querySelector<HTMLButtonElement>(`[data-tabid="${tabId}"]`);
+    button?.focus();
+  }, []);
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/shared/tabbed-section/tabbed-section.tsx]: Fixed in 5421cde. Changed focusTab to use getElementById with the constructed tabDomId instead of querySelector with a CSS selector, eliminating the injection vulnerability.

File: pages/src/components/shared/tabbed-section/tabbed-section.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import classNames from "classnames";
+import { Preconditions } from "../../../base/preconditions";
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/shared/tabbed-section/tabbed-section.tsx]: Fixed in 5421cde. Updated Preconditions import to use @guilty-spark/shared/base/preconditions for consistency with the rest of the pages codebase and to avoid the local re-export indirection.

File: pages/src/components/shared/tabbed-section/tabbed-section.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+      button?.focus();
+    },
+    [tabSetId],
+  );
+
+  const handleKeyDown = React.useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>): void => {
+      if (tabs.length === 0) {
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/shared/tabbed-section/tabbed-section.tsx:36]: Fixed in f60f471. Added a guard to check ctrlKey, altKey, and metaKey and return early if any are held, preserving OS/browser shortcuts like Alt+ArrowLeft for back navigation. Added 3 tests verifying each modifier key is properly ignored.
